src/runtime/printer.h

```diff
@@ -89,6 +89,29 @@ class Printer {
         return *this;
     }
 
+    Printer &operator<<(const halide_type_t &t) {
+        switch(t.code) {
+        case halide_type_int:
```

Comment:
I'd consider assigning the string to a local and making one call to halide_string_to_string.